From bf1c96b92925d50f2d2f993498facea1f87bcfba Mon Sep 17 00:00:00 2001
From: David Kilp <David.Kilp@crossmatch.com>
Date: Tue, 9 Oct 2018 16:15:48 -0400
Subject: [PATCH 04/18] Add device tree support for Crossmatch Sentry (codename
 claimcheck) devices:   - rev1 is imx6q-claimcheck.dtb   - rev2 is
 imx6q-claimcheck-ams.dtb

---
 arch/arm/boot/dts/Makefile                 |    2 +
 arch/arm/boot/dts/imx6q-claimcheck-ams.dts |   53 ++
 arch/arm/boot/dts/imx6q-claimcheck.dts     |   55 ++
 arch/arm/boot/dts/imx6qdl-claimcheck.dtsi  | 1357 ++++++++++++++++++++++++++++
 4 files changed, 1467 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx6q-claimcheck-ams.dts
 create mode 100644 arch/arm/boot/dts/imx6q-claimcheck.dts
 create mode 100644 arch/arm/boot/dts/imx6qdl-claimcheck.dtsi

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 4b68995..156de54 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -435,6 +435,8 @@ dtb-$(CONFIG_SOC_IMX6Q) += \
 	imx6q-utilite-pro.dtb \
 	imx6q-wandboard.dtb \
 	imx6q-wandboard-revb1.dtb \
+	imx6q-claimcheck.dtb \
+	imx6q-claimcheck-ams.dtb \
 	imx6qp-nitrogen6_max.dtb \
 	imx6qp-sabreauto.dtb \
 	imx6qp-sabreauto-ecspi.dtb \
diff --git a/arch/arm/boot/dts/imx6q-claimcheck-ams.dts b/arch/arm/boot/dts/imx6q-claimcheck-ams.dts
new file mode 100644
index 0000000..db26c0d
--- /dev/null
+++ b/arch/arm/boot/dts/imx6q-claimcheck-ams.dts
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2016 Cross Match Technologies.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ * 
+ * This is the device tree file override for a Sentry/Claimcheck unit
+ * for Hardware board revision 1.1 
+ *
+ * The revision 1.1 board has the AMS AS3911 NFC 
+ * controller used in place of the older NXP PN544 NFC controller.
+ *
+ * With current AMS NFC controller firmware you cannot remove power to
+ * the chip when going to suspend/sleep operations as it expects the 
+ * state of the chip to remain the same.
+ * 
+ * The chip power is controlled by GPIOs for SC_USB_EN (GPIO-81) and 
+ * Aux-5V (GPIO-169). Therefore these MUST remain high at all times --
+ * even during sleep (which will impact the battery life).
+ *
+ * The normal Sentry power management turns these 2 lines off through
+ * both the Aux-5V device (cmtaux5v) and the pcsc daemon directly 
+ * controlling the SC_USB_EN line through a virtual GPIO at:
+ * /sys/devices/virtual/sec/smartcard/smartcard_enable/
+ * 
+ * For rev 1.1 boards the smartcard_enable is re-mapped to the SC_INT
+ * (GPIO-175) which is now tied to the reset pin of the contact card
+ * reader(s) so in effect does the same as providing a switch to control
+ * power to the readers as it was doing before.
+ *
+ * This file just needs to "overlay" the normal claimcheck dts file to
+ * 
+ * 1. Tell the cmtaux5v driver NOT to turn of power to the Aux-5V during
+ *    suspend operations by setting the "use_pm" value to 0.
+ *
+ * 2. Re-map the power control for the smartcard reader(s) from SC_USB_EN
+ *    to the SC_INT line 
+ */
+
+#include "imx6q-claimcheck.dts"
+
+
+/ {
+
+	/* Power control -OR- Reset GPIO to contact smartcard reader(s) */
+	smartcard_ctrl {
+		sc_power-on-gpio = <&gpio6 15 GPIO_ACTIVE_HIGH>;	/* SC_INT - for rev 1.1 boards this is connected to reader(s) reset */
+		power-on-gpio = <&gpio3 17 GPIO_ACTIVE_HIGH>;		/* SC_USB_EN - for rev 1.1 boards this supplies power to BOTH smarcard + NFC */
+	};
+		
+};
+
diff --git a/arch/arm/boot/dts/imx6q-claimcheck.dts b/arch/arm/boot/dts/imx6q-claimcheck.dts
new file mode 100644
index 0000000..7f406eb
--- /dev/null
+++ b/arch/arm/boot/dts/imx6q-claimcheck.dts
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2015 Cross Match, Inc.
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+/dts-v1/;
+
+#include "imx6q.dtsi"
+#include "imx6qdl-claimcheck.dtsi"
+
+/ {
+	model = "Freescale i.MX6 Quad Cross Match Claimcheck Board";
+	compatible = "fsl,imx6q-claimcheck", "fsl,imx6q";
+};
+
+&reg_arm {
+	/delete-property/ vin-supply;
+};
+
+&reg_pu {
+	/delete-property/ vin-supply;
+};
+
+&reg_soc {
+	/delete-property/ vin-supply;
+};
+
+&mxcfb1 {
+	status = "okay";
+};
+
+&mxcfb2 {
+	status = "okay";
+};
+
+&mxcfb3 {
+	status = "okay";
+};
+
+&mxcfb4 {
+	status = "okay";
+};
+
+&sata {
+	status = "disabled";
+};
+
diff --git a/arch/arm/boot/dts/imx6qdl-claimcheck.dtsi b/arch/arm/boot/dts/imx6qdl-claimcheck.dtsi
new file mode 100644
index 0000000..09dba2d
--- /dev/null
+++ b/arch/arm/boot/dts/imx6qdl-claimcheck.dtsi
@@ -0,0 +1,1357 @@
+/*
+ * Copyright 2018 Cross Match Technologies, Inc.
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/power/ltc4155.h>
+
+/ {
+	aliases {
+		mxcfb0 = &mxcfb1;
+		mxcfb1 = &mxcfb2;
+		mxcfb2 = &mxcfb3;
+		mxcfb3 = &mxcfb4;
+	};
+
+	/* dkilp: this is not a device but serves as a place for 
+	 * passing data between firmware and the operating system, 
+	 * like boot arguments. In this case this is for boot console
+	 * output for kernel crash dumps.
+	 */
+	chosen {
+		stdout-path = &uart2;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		/* 
+		 * The user Red/Green LED is under control of Android.
+		 * Do not change the name as they match what Android expects
+		 * and configures the /sys/class/led/xxx permissions correctly.
+		 * 
+		 * We currently don't support blinking but we could 
+		 * configure with trigger = timer so we can make them
+		 * blink if requested.
+		 */
+		red-led {
+			label = "red";
+			gpios = <&gpio3 3 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "none";
+			default-state = "off";
+		};
+		
+		green-led {
+			label = "green";
+			gpios = <&gpio1 2 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "none";
+			default-state = "off";
+		};
+		
+		/* these are LEDs for board debugging and aren't visible when unit is assembled */
+		debug-led {
+			label = "heartbeat-led";
+			gpios = <&gpio3 8 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "heartbeat";
+			default-state = "off";
+		};
+
+		mmc-led {
+			label = "mmc-led";
+			gpios = <&gpio3 9 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "mmc1";
+			default-state = "off";
+		};
+			
+	};
+
+	memory: memory {
+		reg = <0x10000000 0x40000000>;
+	};
+
+	regulators {
+		compatible = "simple-bus";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		reg_usb_otg_vbus: regulator@0 {
+			compatible = "regulator-fixed";
+			reg = <0>;
+			regulator-name = "usb_otg_vbus";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			gpio = <&gpio3 22 0>;
+			enable-active-high;
+			vin-supply = <&swbst_reg>;
+		};
+
+		reg_usb_h1_vbus: regulator@1 {
+			compatible = "regulator-fixed";
+			reg = <1>;
+			regulator-name = "usb_h1_vbus";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			gpio = <&gpio3 31 0>;
+			enable-active-high;
+			vin-supply = <&swbst_reg>;
+		};
+
+		reg_audio: regulator@2 {
+			compatible = "regulator-fixed";
+			reg = <2>;
+			regulator-name = "wm8962-supply";
+			regulator-min-microvolt = <4200000>;
+			regulator-max-microvolt = <4200000>;
+			gpio = <&gpio4 10 0>;
+			enable-active-high;
+		};
+
+#if 0
+/* On SabreSD this is the Display Power Enable line. Claimcheck has this as Elektra power enable*/
+		reg_mipi_dsi_pwr_on: mipi_dsi_pwr_on {
+			compatible = "regulator-fixed";
+			regulator-name = "mipi_dsi_pwr_on";
+			gpio = <&gpio6 14 0>;
+			enable-active-high;
+		};
+#endif
+
+		reg_sensor: regulator@3 {
+			compatible = "regulator-fixed";
+			reg = <3>;
+			regulator-name = "sensor-supply";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			gpio = <&gpio2 31 0>;
+			startup-delay-us = <500>;
+			enable-active-high;
+		};
+
+		reg_aux5v: regulator@4 {
+			compatible = "regulator-fixed";
+			reg = <4>;
+			regulator-name = "aux5v-supply";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			gpio = <&gpio6 9 0>;
+			enable-active-high;
+		};
+
+		reg_wlan_en: regulator@5 {
+			compatible = "regulator-fixed";
+			reg = <5>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&pinctrl_reg_wlan_en>;
+			regulator-name = "wlan-en";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			gpio = <&gpio3 27 GPIO_ACTIVE_HIGH>;
+			startup-delay-us = <70000>;
+			enable-active-high;
+		};
+
+#if 0
+		reg_usb_h2_vbus: regulator@6 {
+			compatible = "regulator-fixed";
+			reg = <6>;
+			regulator-name = "usb_h2_vbus";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			gpio = <&gpio6 8 0>;
+			enable-active-high;
+		};
+#endif
+
+		reg_hdmi: regulator@6 {
+			compatible = "regulator-fixed";
+			reg = <6>;
+			regulator-name = "hdmi-5v-supply";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			enable-active-high;
+			hdmi-5v-supply = <&swbst_reg>;
+		};
+
+	};
+
+	gpio-keys {
+		compatible = "gpio-keys";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_gpio_keys>;
+
+		power {
+			label = "Power Button";
+			gpios = <&gpio3 29 GPIO_ACTIVE_LOW>;
+			wakeup-source;
+			linux,code = <KEY_POWER>;
+		};
+
+		volume-up {
+			label = "Volume Up";
+			gpios = <&gpio1 4 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_VOLUMEUP>;
+		};
+
+		volume-down {
+			label = "Volume Down";
+			gpios = <&gpio1 5 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_VOLUMEDOWN>;
+		};
+	};
+
+	sound {
+		compatible = "fsl,imx6q-sabresd-wm8962",
+			   "fsl,imx-audio-wm8962";
+		model = "wm8962-audio";
+		cpu-dai = <&ssi2>;
+		audio-codec = <&codec>;
+		asrc-controller = <&asrc>;
+		audio-routing =
+			"Headphone Jack", "HPOUTL",
+			"Headphone Jack", "HPOUTR",
+			"Ext Spk", "SPKOUTL",
+			"Ext Spk", "SPKOUTR",
+			"AMIC", "MICBIAS",
+			"IN3R", "AMIC",
+			"DMIC", "MICBIAS",
+			"DMICDAT", "DMIC",
+			"CPU-Playback", "ASRC-Playback",
+			"Playback", "CPU-Playback",
+			"ASRC-Capture", "CPU-Capture",
+			"CPU-Capture", "Capture";
+		mux-int-port = <2>;
+		mux-ext-port = <3>;
+		codec-master;
+		hp-det-gpios = <&gpio7 8 0>;
+		mic-det-gpios = <&gpio5 20 1>;
+	};
+
+	sound-hdmi {
+		compatible = "fsl,imx6q-audio-hdmi",
+			     "fsl,imx-audio-hdmi";
+		model = "imx-audio-hdmi";
+		hdmi-controller = <&hdmi_audio>;
+	};
+
+	mxcfb1: fb@0 {
+		compatible = "fsl,mxc_sdc_fb";
+		disp_dev = "lcd";
+		interface_pix_fmt = "RGB24";
+		mode_str ="SHARP-WVGA";
+		default_bpp = <32>;
+		int_clk = <0>;
+		late_init = <0>;
+		status = "disabled";
+	};
+
+	mxcfb2: fb@1 {
+		compatible = "fsl,mxc_sdc_fb";
+		disp_dev = "hdmi";
+		interface_pix_fmt = "RGB24";
+		mode_str ="1920x1080M@60";
+		default_bpp = <24>;
+		int_clk = <0>;
+		late_init = <0>;
+		status = "disabled";
+	};
+
+	mxcfb3: fb@2 {
+		compatible = "fsl,mxc_sdc_fb";
+		disp_dev = "ldb";
+		interface_pix_fmt = "RGB666";
+		default_bpp = <16>;
+		int_clk = <0>;
+		late_init = <0>;
+		status = "disabled";
+	};
+
+	mxcfb4: fb@3 {
+		compatible = "fsl,mxc_sdc_fb";
+		disp_dev = "ldb";
+		interface_pix_fmt = "RGB666";
+		default_bpp = <16>;
+		int_clk = <0>;
+		late_init = <0>;
+		status = "disabled";
+	};
+
+	lcd@0 {
+		compatible = "fsl,lcd";
+		ipu_id = <0>;
+		disp_id = <0>;
+		default_ifmt = "RGB565";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_ipu1>;
+		reset-gpio = <&gpio5 4 GPIO_ACTIVE_LOW>;
+		status = "okay";
+	};
+
+	pwm-backlight {
+		compatible = "pwm-backlight";
+		pwms = <&pwm1 0 5000000>;
+		brightness-levels = <
+			0  1  2  3  4  5  6  7  8  9
+			10 11 12 13 14 15 16 17 18 19
+			20 21 22 23 24 25 26 27 28 29
+			30 31 32 33 34 35 36 37 38 39
+			40 41 42 43 44 45 46 47 48 49
+			50 51 52 53 54 55 56 57 58 59
+			60 61 62 63 64 65 66 67 68 69
+			70 71 72 73 74 75 76 77 78 79
+			80 81 82 83 84 85 86 87 88 89
+			90 91 92 93 94 95 96 97 98 99
+			100 101 102 103 104 105 106 107 108 109
+			110 111 112 113 114 115 116 117 118 119
+			120 121 122 123 124 125 126 127 128 129
+			130 131 132 133 134 135 136 137 138 139
+			140 141 142 143 144 145 146 147 148 149
+			150 151 152 153 154 155 156 157 158 159
+			160 161 162 163 164 165 166 167 168 169
+			170 171 172 173 174 175 176 177 178 179
+			180 181 182 183 184 185 186 187 188 189
+			190 191 192 193 194 195 196 197 198 199
+			200 201 202 203 204 205 206 207 208 209 
+			210 211 212 213 214 215 216 217 218 219 
+			220 221 222 223 224 225 226 227 228 229 
+			230 231 232 233 234 235 236 237 238 239 
+			240 241 242 243 244 245 246 247 248 249
+			250 251 252 253 254 255
+			>;
+		default-brightness-level = <102>;
+		status = "okay";
+	};
+
+	/* Elektra parallel CSI=0 */
+	v4l2_cap_0 {
+		compatible = "fsl,imx6q-v4l2-capture";
+		ipu_id = <0>;
+		csi_id = <0>;
+		mclk_source = <0>;
+		status = "okay";
+	};
+
+	/* OV5640 MIPI CSI=1 */
+	v4l2_cap_1 {
+		compatible = "fsl,imx6q-v4l2-capture";
+		ipu_id = <0>;
+		csi_id = <1>;
+		mclk_source = <0>;
+		status = "okay";
+	};
+
+	v4l2_out {
+		compatible = "fsl,mxc_v4l2_output";
+		status = "okay";
+	};
+
+	btwilink {
+		compatible = "btwilink";
+	};
+
+	kim {
+		compatible = "kim";
+		nshutdown_gpio = <128>;  // GPIO5_0 The wl8 driver expects gpio to be an integer, so gpio5_0 is (5-1)*32+0=128
+		dev_name = "/dev/ttymxc4";
+		flow_cntrl = <1>;
+		baud_rate = <3000000>;
+	};
+
+	caam_keyblob {
+		compatible = "fsl,sec-v4.0-keyblob";
+	};
+
+	cmtaux5v {
+		compatible = "cmt,cmtaux5v-gpio";
+		aux5v-supply = <&reg_aux5v>;
+		use_pm = <0>;	/* set this true(1) if we want Power Manager to suspend/resume this device */
+	};
+
+	cmtgps {
+		compatible = "cmt,cmtgps-gpio";
+		pwn-gpios = <&gpio3 0 GPIO_ACTIVE_HIGH>;
+	};
+	
+	cmtusbhostport {
+		compatible = "cmt,cmtusbhostport-gpio";
+		port-gpios = <&gpio6 8 GPIO_ACTIVE_HIGH>;
+		use_pm = <1>;	/* set this true(1) if we want Power Manager to suspend/resume this device */
+	};
+
+	/* Power control -OR- Reset GPIO to contact smartcard reader(s) */
+	smartcard_ctrl {
+		sc_power-on-gpio = <&gpio3 17 GPIO_ACTIVE_HIGH>;	/* SC_USB_EN */
+	};
+	
+	/* Battery Remove Detect/Reset gpios */
+	battery_ctrl {
+		battery_removed-gpio = <&gpio3 20 GPIO_ACTIVE_HIGH>;	
+		battery_removed_reset-gpio = <&gpio3 21 GPIO_ACTIVE_HIGH>;	
+	};
+
+	/* Battery Charger LED (under control of charger) */
+	led_charger_ctrl {
+		led_charger-gpio = <&gpio7 12 GPIO_ACTIVE_HIGH>;	
+	};
+
+};
+
+&audmux {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_audmux>;
+	status = "okay";
+};
+
+&ecspi2 {
+	fsl,spi-num-chipselects = <2>;
+	cs-gpios = <&gpio2 26 0>,
+			<&gpio2 27 0>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_ecspi2>;
+	status = "okay";
+
+	flash: m25p80@0 {
+		compatible = "st,m25p64";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		spi-max-frequency = <20000000>;
+		reg = <0>;
+		m25p,fast-read;
+
+		/* Elektra partition using all 8 MB */
+		mtd@00000000 {
+			label = "elektra";
+			reg = <0x00000000 0x800000>;
+		};
+		
+	};
+};
+
+&fec {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_enet>;
+	phy-supply = <&swbst_reg>;
+	phy-mode = "rmii";
+	fsl,magic-packet;
+	status = "disabled";
+};
+
+&i2c1 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c1>;
+	status = "okay";
+
+	/*
+	 * LTC4155 - Dual-Input Charger Config
+	 * there are separate settings for the USB and A/C chargers
+	 * 
+	 * claimcheck changes the default settings for:
+	 *  - battery charger float voltage from 4.05 to 4.20 V
+	 *  - Limit battery charger current limit to 43.75% to limit heat build up on Rev2 boards
+	 *  - No Limit (100%) on battery charger current for Rev3 boards
+	 */
+	ltc4155@09 {
+		compatible = "ltc,ltc4155";
+		reg = <0x09>;
+		interrupt-parent = <&gpio3>;
+		interrupts = <23 IRQ_TYPE_EDGE_FALLING>;
+		debounce_delay = <10>;		/* debounce time in msec */
+		eoc_level_ac = <LTC4155_EOC_10P>;
+		ilimit_ac = <LTC4155_ICHG_100P>;
+		vloat_ac = <LTC4155_EOC_4_20V>;
+		eoc_level_usb = <LTC4155_EOC_10P>;
+		ilimit_usb = <LTC4155_ICHG_100P>;
+		vloat_usb = <LTC4155_EOC_4_20V>;
+	};
+
+	codec: wm8962@1a {
+		compatible = "wlf,wm8962";
+		reg = <0x1a>;
+		clocks = <&clks IMX6QDL_CLK_CKO>;
+		DCVDD-supply = <&reg_audio>;
+		DBVDD-supply = <&reg_audio>;
+		AVDD-supply = <&reg_audio>;
+		CPVDD-supply = <&reg_audio>;
+		MICVDD-supply = <&reg_audio>;
+		PLLVDD-supply = <&reg_audio>;
+		SPKVDD1-supply = <&reg_audio>;
+		SPKVDD2-supply = <&reg_audio>;
+		/* spk-mono;  this does not work correctly! */
+		amic-mono;
+		dmic-mono;
+		mic-cfg = <0x0200>;
+		gpio-cfg = <
+			0x0000 /* 0:Default */
+			0x0000 /* 1:Default */
+			0x0013 /* 2:FN_DMICCLK */
+			0x0000 /* 3:Default */
+			0x8014 /* 4:FN_DMICCDAT */
+			0x0416 /* 5:FN_MICD */
+		>;
+       };
+
+	mma8451@1c {
+		compatible = "fsl,mma8451";
+		reg = <0x1c>;
+		position = <0>;
+		vdd-supply = <&reg_sensor>;
+		vddio-supply = <&reg_sensor>;
+		interrupt-parent = <&gpio1>;
+		interrupts = <18 8>;
+		interrupt-route = <1>;
+	};
+
+	/*
+	 * NOTE: 
+	 *	VGEN2 supply is not needed on Claimcheck as we using a 1.8V I/O (DOVDD)
+	 *  and the manual says we should use the internal DVDD regulator and do not
+	 *  supply an external DVDD source. This says ~25 mA of standby current.
+	*/
+	ov564x_mipi: ov564x_mipi@3c { /* i2c2 driver */
+		compatible = "ovti,ov564x_mipi";
+		reg = <0x3c>;
+		clocks = <&clks 201>;
+		clock-names = "csi_mclk";
+		DOVDD-supply = <&vgen4_reg>; /* 1.8v */
+		AVDD-supply = <&vgen3_reg>;  /* 2.8v, VGEN3 in u-boot */
+		/* DVDD-supply = <&vgen2_reg>; */ /* 1.5v */
+		pwn-gpios = <&gpio1 19 GPIO_ACTIVE_LOW>; 	/* Power DOWN pin active low: SD1_DAT2 */
+		rst-gpios = <&gpio1 20 GPIO_ACTIVE_HIGH>;   /* reset SD1_CLK */
+		csi_id = <1>;
+		mclk = <24000000>;
+		mclk_source = <0>;
+		flash-gpio = <&gpio2 1 GPIO_ACTIVE_HIGH>; 		/* Flash led */
+		flash_en-gpio = <&gpio2 2 GPIO_ACTIVE_HIGH>; 	/* Flash enable/Torch led */
+		flash_type-gpio = <&gpio2 4 GPIO_ACTIVE_HIGH>; 	/* Flash mugshot/credential */
+	};
+
+};
+
+&i2c2 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c2>;
+	status = "okay";
+
+	hdmi_edid: edid@50 {
+		compatible = "fsl,imx6-hdmi-i2c";
+		reg = <0x50>;
+	};
+
+	pmic: pfuze100@08 {
+		compatible = "fsl,pfuze100";
+		reg = <0x08>;
+
+		regulators {
+			sw1a_reg: sw1ab {
+				regulator-min-microvolt = <300000>;
+				regulator-max-microvolt = <1875000>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
+
+			sw1c_reg: sw1c {
+				regulator-min-microvolt = <300000>;
+				regulator-max-microvolt = <1875000>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
+
+			sw2_reg: sw2 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
+
+			sw3a_reg: sw3a {
+				regulator-min-microvolt = <400000>;
+				regulator-max-microvolt = <1975000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			sw3b_reg: sw3b {
+				regulator-min-microvolt = <400000>;
+				regulator-max-microvolt = <1975000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			sw4_reg: sw4 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <3300000>;
+			};
+
+			swbst_reg: swbst {
+				regulator-min-microvolt = <5000000>;
+				regulator-max-microvolt = <5150000>;
+			};
+
+			snvs_reg: vsnvs {
+				regulator-min-microvolt = <1000000>;
+				regulator-max-microvolt = <3000000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			vref_reg: vrefddr {
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			vgen1_reg: vgen1 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <1550000>;
+			};
+
+			vgen2_reg: vgen2 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <1550000>;
+			};
+
+			vgen3_reg: vgen3 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+			};
+
+			vgen4_reg: vgen4 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vgen5_reg: vgen5 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vgen6_reg: vgen6 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+			};
+		};
+	};
+
+	/* Fairchild FSA9485 USB port switch */
+	fsa9485: fsa9485@25 {
+		compatible = "fairchild,fsa9485";
+		reg = <0x25>;
+		interrupt-parent = <&gpio3>;
+		interrupts = <16 2>;
+	};
+
+	/* TI BQ27742 Battery guage */
+	bq27742: bq27742@55 {
+		compatible = "ti,bq27742";
+		reg = <0x55>;
+		battery-name = "battery";
+	};
+
+};
+
+&i2c3 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c3>;
+	status = "okay";
+
+	eeprom@52 {
+		compatible = "atmel,24c128";
+		reg = <0x52>;
+		pagesize = <64>;
+	};
+
+	smsc4606@2d {
+		compatible = "smsc,smsc4606";
+		reg = <0x2d>;
+		reset-gpios = <&gpio1 7 GPIO_ACTIVE_LOW>;  	/* Hub reset GPIO_7__GPIO1_IO07 */
+	};
+
+	goodix_ts@5d {
+		compatible = "goodix,gt9xx";
+		reg = <0x5d>;
+		interrupt-parent = <&gpio3>;
+		interrupts = <26 2>;
+#if 0
+		vdd_ana-supply = <&pm8226_l19>;
+		vcc_i2c-supply = <&pm8226_lvs1>;
+#endif
+		goodix,rst-gpio = <&gpio4 20 0x00>;
+		goodix,irq-gpio = <&gpio3 26 0x00>;
+#if 0
+		goodix,cfg-group1 = [
+			41 E0 01 20 03 0A 04 00 01 08 
+			28 0D 50 32 03 05 00 00 00 00 
+			00 00 00 00 00 00 00 87 28 0A 
+			4A 47 31 0D 00 00 01 B9 03 25 
+			00 00 00 00 00 00 00 00 00 00 
+			00 1D 94 94 C5 02 07 00 00 04 
+			E1 22 00 AD 2F 00 87 42 00 70 
+			5B 00 67 7F 00 67 00 00 00 00 
+			00 00 00 00 00 00 00 00 00 00 
+			00 00 00 00 00 00 00 00 00 00 
+			00 00 00 00 00 00 00 00 00 00 
+			00 00 14 12 10 0E 0C 0A 08 06 
+			04 02 FF FF FF FF 00 00 00 00 
+			00 00 00 00 00 00 00 00 00 00 
+			00 00 26 24 22 21 20 1F 1E 1D 
+			0C 0A 08 06 04 02 00 FF FF FF 
+			FF FF FF FF FF FF FF FF 00 00 
+			00 00 00 00 00 00 00 00 00 00 
+			00 00 00 00 14 01];
+#endif
+
+	};
+
+	isl29023@44 {
+		compatible = "fsl,isl29023";
+		reg = <0x44>;
+		rext = <499>;
+		vdd-supply = <&reg_sensor>;
+		interrupt-parent = <&gpio2>;
+		interrupts = <22 2>;
+	};
+
+	/* Elektra Finger sensor on Parallel CSI 0 input */
+	elektra: elektra@42 {
+		compatible = "cmt,elektra";
+		reg = <0x42>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_ipu1_2>;
+		clocks = <&clks IMX6QDL_CLK_CKO>;
+		clock-names = "csi_mclk";
+		pwn-gpios = <&gpio6 14 GPIO_ACTIVE_LOW>;  	/* power Down pin: NANDF_CS1/FPS_PWR_EN */
+		rst-gpios = <&gpio6 11 GPIO_ACTIVE_LOW>;  	/* NANDF_CS0/FPS_RST_B */
+		csi_id = <0>;
+		mclk = <24000000>;
+		mclk_source = <0>;
+	};
+
+};
+
+&iomuxc {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hog>;
+	
+	imx6q-claimcheck {
+	
+		pinctrl_hog: hoggrp {
+			fsl,pins = <
+				MX6QDL_PAD_EIM_D31__GPIO3_IO31	0x80000000		/* HOST_USB_EN  */
+				MX6QDL_PAD_NANDF_CS0__GPIO6_IO11 0x00017059		/* Finger Sensor Reset - FPS_RST_B */
+				MX6QDL_PAD_NANDF_CS1__GPIO6_IO14 0x00017059		/* Finger Sensor Power - FPS_PWR_EN */
+				MX6QDL_PAD_EIM_EB3__GPIO2_IO31 	0x80000000		/* Sensor Power Enable */
+				MX6QDL_PAD_SD3_RST__GPIO7_IO08	0x000000b0		/* Headphone Detect */
+				MX6QDL_PAD_CSI0_DATA_EN__GPIO5_IO20		0x80000000	/* Microphone Detect */
+				MX6QDL_PAD_SD1_CMD__GPIO1_IO18 	0x80000000		/* ACCL_INT_IN */
+				MX6QDL_PAD_SD1_DAT0__GPIO1_IO16 	0x80000000		/* ACCL_INT2_IN */
+				MX6QDL_PAD_NANDF_WP_B__GPIO6_IO09	0x0001b0b0		/* Aux 5V Enabled */
+
+				MX6QDL_PAD_EIM_A24__GPIO5_IO04	0x0001b0b0		/* LCD Display Reset */
+				MX6QDL_PAD_NANDF_D0__GPIO2_IO00 0x80000000		/* SD3_CD_B */
+
+				MX6QDL_PAD_NANDF_D1__GPIO2_IO01 0x0001b0b0		/* FLASH */
+				MX6QDL_PAD_NANDF_D2__GPIO2_IO02 0x0001b0b0		/* FLASH_EN */
+				MX6QDL_PAD_NANDF_D3__GPIO2_IO03 0x80000000		/* TP14 */
+				MX6QDL_PAD_NANDF_D4__GPIO2_IO04	0x0001b0b0		/* FLASH_LED_SEL */
+				MX6QDL_PAD_NANDF_D5__GPIO2_IO05	0x80000000		/* PMIC_INT_B */
+				MX6QDL_PAD_NANDF_D7__GPIO2_IO07	0x80000000		/* SPARE_GPIO */
+				MX6QDL_PAD_EIM_A18__GPIO2_IO20	0x000100b0		/* MODEM_EN */
+				MX6QDL_PAD_EIM_A19__GPIO2_IO19	0x000100b0		/* MODEM_E_0 */
+				MX6QDL_PAD_EIM_A20__GPIO2_IO18	0x000100b0		/* MODEM_IGNITION */
+				MX6QDL_PAD_GPIO_0__CCM_CLKO1    0x00017099		/* CSI_MCLK */
+				MX6QDL_PAD_EIM_DA0__GPIO3_IO00	0x000100b0		/* GPS_PWREN */
+				MX6QDL_PAD_EIM_A16__GPIO2_IO22	0x80000000		/* Ambient Light Sensor IRQ ALS_INT */
+				MX6QDL_PAD_SD1_CLK__GPIO1_IO20	0x000170f1		/* Camera Reset, CSI_RST_B */
+				MX6QDL_PAD_SD1_DAT2__GPIO1_IO19	0x000170f1		/* Camera GP CSI_PWN (Camera Powerdown) */
+				MX6QDL_PAD_SD1_DAT3__GPIO1_IO21	0x00013059		/* WLAN_IRQ configure as an input with a pull down */
+				MX6QDL_PAD_EIM_D23__GPIO3_IO23 0x000120b0		/* CHARGER_IRQ LTC4155 Interrupt */
+				MX6QDL_PAD_EIM_D16__GPIO3_IO16	0x0001b000		/* Fairchild FSA9485 USB Port Switch IRQ - weak pull-up */
+				MX6QDL_PAD_NANDF_RB0__GPIO6_IO10	0x0001b0b0	/* NFC_EN */
+				MX6QDL_PAD_EIM_A17__GPIO2_IO21	0x000130b0		/* NFC_INT */
+				MX6QDL_PAD_EIM_BCLK__GPIO6_IO31	0x0000b0b1		/* NFC_DWNLD */
+				MX6QDL_PAD_EIM_D17__GPIO3_IO17	0x0001b0b0		/* SC_USB_EN: smart card enable (USB Power) */
+				MX6QDL_PAD_NANDF_CS2__GPIO6_IO15	0x0001b0b0	/* SC_INT: on Rev 1.1 this is connected to smartcard reader(s) reset */
+				MX6QDL_PAD_EIM_D18__GPIO3_IO18	0x000000b0		/* GPS_PPS (input) */
+				MX6QDL_PAD_EIM_D20__GPIO3_IO20	0x000120b0		/* BAT_RM_SENSE (battery remove sense input). Configure as for IRQ with no pull-ups */
+				MX6QDL_PAD_EIM_D21__GPIO3_IO21	0x0001b0b0		/* BAT_RM_RST (battery remove reset output) */
+				MX6QDL_PAD_NANDF_CLE__GPIO6_IO07 0x0001b0b0		/* FP_ENABLE (rev-3) was Microphone Detect (broken on rev-2) */
+				MX6QDL_PAD_EIM_A25__GPIO5_IO02 0x80000000		/* Charge_Complete */
+				MX6QDL_PAD_SD1_DAT1__GPIO1_IO17	0x80000000		/* BT_WAKE */
+				MX6QDL_PAD_NANDF_CS3__GPIO6_IO16 0x000000b0		/* HOST2_USB_OCb configure as Input: no pull-ups*/
+				MX6QDL_PAD_GPIO_7__GPIO1_IO07	0x80000000		/* SMSC USB Hub Reset */
+				MX6QDL_PAD_RGMII_TXC__XTALOSC_REF_CLK_24M	0x00013030	/* 24MHz ref clock out */
+				MX6QDL_PAD_KEY_COL2__GPIO4_IO10	0x0001b0b0		/* CODEC_PWR_EN */
+				MX6QDL_PAD_DI0_PIN4__GPIO4_IO20	0x80000000		/* LCD_TS_RST: touchscreen reset */
+				MX6QDL_PAD_EIM_D26__GPIO3_IO26 0x80000000		/* touchscreen IRQ */
+				MX6QDL_PAD_NANDF_ALE__GPIO6_IO08 0x80000000		/* HOST2_USB_EN */
+				MX6QDL_PAD_EIM_D22__GPIO3_IO22  0x80000000		/* USB OTG PWR Enable */
+				MX6QDL_PAD_EIM_D30__USB_H1_OC	0x80000000		/* Not used */	
+				
+				/* LEDS */
+				MX6QDL_PAD_EIM_DA8__GPIO3_IO08	0x80000000		/* LED D18 (u-boot) Green */
+				MX6QDL_PAD_GPIO_2__GPIO1_IO02	0x80000000		/* LED D1 USR_LED D1-GREEN/red*/
+				MX6QDL_PAD_EIM_DA3__GPIO3_IO03	0x80000000		/* LED D1 LED D17 (user) D1-RED/green*/
+				MX6QDL_PAD_EIM_DA9__GPIO3_IO09	0x80000000		/* LED D19 (kernel) Green */
+				MX6QDL_PAD_GPIO_17__GPIO7_IO12	0x80000000		/* LED D33 Green/Red Battery Charger LED Enable (Rev-3 boards) / TP-11 (rev-2)  */
+				
+			>;
+		};
+
+		pinctrl_audmux: audmuxgrp {
+			fsl,pins = <
+				MX6QDL_PAD_CSI0_DAT7__AUD3_RXD		0x130b0
+				MX6QDL_PAD_CSI0_DAT4__AUD3_TXC		0x130b0
+				MX6QDL_PAD_CSI0_DAT5__AUD3_TXD		0x110b0
+				MX6QDL_PAD_CSI0_DAT6__AUD3_TXFS		0x130b0
+			>;
+		};
+
+		pinctrl_reg_wlan_en: reg-wlan-engrp {
+			fsl,pins = <
+				MX6QDL_PAD_EIM_D27__GPIO3_IO27		0x030b0		/* WLAN_EN wl12xx_wl_en */
+			>;
+		};
+
+		pinctrl_ecspi2: ecspi2grp {
+			fsl,pins = <
+				MX6QDL_PAD_EIM_OE__ECSPI2_MISO		0x0000b0b1
+				MX6QDL_PAD_EIM_CS1__ECSPI2_MOSI		0x0000b0b1
+				MX6QDL_PAD_EIM_CS0__ECSPI2_SCLK		0x0000b0b1
+				MX6QDL_PAD_EIM_RW__GPIO2_IO26		0x0000b0b1	/* SPI2_CS0: chip select for Serial Flash */
+				MX6QDL_PAD_EIM_LBA__GPIO2_IO27		0x0000b0b1	/* SPI2_CS1: chip select for Sharp display */	
+			>;
+		};
+
+		pinctrl_enet: enetgrp {
+			fsl,pins = <
+				MX6QDL_PAD_ENET_MDIO__ENET_MDIO		0x1b0b0
+				MX6QDL_PAD_ENET_MDC__ENET_MDC		0x1b0b0
+				MX6QDL_PAD_ENET_CRS_DV__ENET_RX_EN	0x1b0b0
+				MX6QDL_PAD_ENET_REF_CLK__ENET_TX_CLK	0x1b0b0		/* not used by RMII interface */
+				MX6QDL_PAD_GPIO_16__ENET_REF_CLK	0x4001b0a8		/* 50 MHz ref clock out */
+				MX6QDL_PAD_ENET_RX_ER__ENET_RX_ER	0x1b0b0
+				MX6QDL_PAD_ENET_TX_EN__ENET_TX_EN	0x1b0b0
+				MX6QDL_PAD_ENET_RXD0__ENET_RX_DATA0	0x1b0b0
+				MX6QDL_PAD_ENET_RXD1__ENET_RX_DATA1	0x1b0b0
+				MX6QDL_PAD_ENET_TXD0__ENET_TX_DATA0	0x1b0b0
+				MX6QDL_PAD_ENET_TXD1__ENET_TX_DATA1	0x1b0b0
+				MX6QDL_PAD_GPIO_18__ENET_RX_CLK		0x1b0b0
+				MX6QDL_PAD_GPIO_19__ENET_TX_ER		0x1b0b0
+			>;
+		};
+
+		pinctrl_gpio_keys: gpio_keysgrp {
+			fsl,pins = <
+				MX6QDL_PAD_EIM_D29__GPIO3_IO29 0x0001b0b0	/* power button */
+				MX6QDL_PAD_GPIO_4__GPIO1_IO04  0x0001b0b0	/* volume up */
+				MX6QDL_PAD_GPIO_5__GPIO1_IO05  0x0001b0b0	/* volume down */
+			>;
+		};
+
+		pinctrl_hdmi_cec: hdmicecgrp {
+			fsl,pins = <
+				MX6QDL_PAD_KEY_ROW2__HDMI_TX_CEC_LINE 0x108b0
+			>;
+		};
+
+		pinctrl_hdmi_hdcp: hdmihdcpgrp {
+			fsl,pins = <
+				MX6QDL_PAD_KEY_COL3__HDMI_TX_DDC_SCL 0x4001b8b1
+				MX6QDL_PAD_KEY_ROW3__HDMI_TX_DDC_SDA 0x4001b8b1
+			>;
+		};
+
+		pinctrl_i2c1: i2c1grp {
+			fsl,pins = <
+				MX6QDL_PAD_CSI0_DAT8__I2C1_SDA		0x4001b8b1
+				MX6QDL_PAD_CSI0_DAT9__I2C1_SCL		0x4001b8b1
+			>;
+		};
+
+		pinctrl_i2c2: i2c2grp {
+			fsl,pins = <
+				MX6QDL_PAD_KEY_COL3__I2C2_SCL		0x4001b8b1
+				MX6QDL_PAD_KEY_ROW3__I2C2_SDA		0x4001b8b1
+			 >;
+		};
+
+		pinctrl_i2c3: i2c3grp {
+			fsl,pins = <
+				MX6QDL_PAD_GPIO_3__I2C3_SCL		0x4001b8b1
+				MX6QDL_PAD_GPIO_6__I2C3_SDA		0x4001b8b1
+			>;
+		};
+
+		pinctrl_ipu1: ipu1grp {
+			fsl,pins = <
+				MX6QDL_PAD_DI0_DISP_CLK__IPU1_DI0_DISP_CLK 0x10
+				MX6QDL_PAD_DI0_PIN15__IPU1_DI0_PIN15       0x10
+				MX6QDL_PAD_DI0_PIN2__IPU1_DI0_PIN02        0x10
+				MX6QDL_PAD_DI0_PIN3__IPU1_DI0_PIN03        0x10
+				MX6QDL_PAD_DISP0_DAT0__IPU1_DISP0_DATA00   0x10
+				MX6QDL_PAD_DISP0_DAT1__IPU1_DISP0_DATA01   0x10
+				MX6QDL_PAD_DISP0_DAT2__IPU1_DISP0_DATA02   0x10
+				MX6QDL_PAD_DISP0_DAT3__IPU1_DISP0_DATA03   0x10
+				MX6QDL_PAD_DISP0_DAT4__IPU1_DISP0_DATA04   0x10
+				MX6QDL_PAD_DISP0_DAT5__IPU1_DISP0_DATA05   0x10
+				MX6QDL_PAD_DISP0_DAT6__IPU1_DISP0_DATA06   0x10
+				MX6QDL_PAD_DISP0_DAT7__IPU1_DISP0_DATA07   0x10
+				MX6QDL_PAD_DISP0_DAT8__IPU1_DISP0_DATA08   0x10
+				MX6QDL_PAD_DISP0_DAT9__IPU1_DISP0_DATA09   0x10
+				MX6QDL_PAD_DISP0_DAT10__IPU1_DISP0_DATA10  0x10
+				MX6QDL_PAD_DISP0_DAT11__IPU1_DISP0_DATA11  0x10
+				MX6QDL_PAD_DISP0_DAT12__IPU1_DISP0_DATA12  0x10
+				MX6QDL_PAD_DISP0_DAT13__IPU1_DISP0_DATA13  0x10
+				MX6QDL_PAD_DISP0_DAT14__IPU1_DISP0_DATA14  0x10
+				MX6QDL_PAD_DISP0_DAT15__IPU1_DISP0_DATA15  0x10
+				MX6QDL_PAD_DISP0_DAT16__IPU1_DISP0_DATA16  0x10
+				MX6QDL_PAD_DISP0_DAT17__IPU1_DISP0_DATA17  0x10
+				MX6QDL_PAD_DISP0_DAT18__IPU1_DISP0_DATA18  0x10
+				MX6QDL_PAD_DISP0_DAT19__IPU1_DISP0_DATA19  0x10
+				MX6QDL_PAD_DISP0_DAT20__IPU1_DISP0_DATA20  0x10
+				MX6QDL_PAD_DISP0_DAT21__IPU1_DISP0_DATA21  0x10
+				MX6QDL_PAD_DISP0_DAT22__IPU1_DISP0_DATA22  0x10
+				MX6QDL_PAD_DISP0_DAT23__IPU1_DISP0_DATA23  0x10
+			>;
+		};
+
+		pinctrl_ipu1_2: ipu1grp-2 { /* parallel camera */
+			fsl,pins = <
+				MX6QDL_PAD_CSI0_DAT10__IPU1_CSI0_DATA10    0x80000000
+				MX6QDL_PAD_CSI0_DAT11__IPU1_CSI0_DATA11    0x80000000
+				MX6QDL_PAD_CSI0_DAT12__IPU1_CSI0_DATA12    0x80000000
+				MX6QDL_PAD_CSI0_DAT13__IPU1_CSI0_DATA13    0x80000000
+				MX6QDL_PAD_CSI0_DAT14__IPU1_CSI0_DATA14    0x80000000
+				MX6QDL_PAD_CSI0_DAT15__IPU1_CSI0_DATA15    0x80000000
+				MX6QDL_PAD_CSI0_DAT16__IPU1_CSI0_DATA16    0x80000000
+				MX6QDL_PAD_CSI0_DAT17__IPU1_CSI0_DATA17    0x80000000
+				MX6QDL_PAD_CSI0_DAT18__IPU1_CSI0_DATA18    0x80000000
+				MX6QDL_PAD_CSI0_DAT19__IPU1_CSI0_DATA19    0x80000000
+				MX6QDL_PAD_CSI0_PIXCLK__IPU1_CSI0_PIXCLK   0x80000000
+				MX6QDL_PAD_CSI0_MCLK__IPU1_CSI0_HSYNC      0x80000000
+				MX6QDL_PAD_CSI0_VSYNC__IPU1_CSI0_VSYNC     0x80000000
+			>;
+		};
+
+		pinctrl_pwm1: pwm1grp {
+			fsl,pins = <
+				MX6QDL_PAD_GPIO_9__PWM1_OUT			0x1b0b1
+			>;
+		};
+
+		pinctrl_uart1: uart1grp {
+			fsl,pins = <
+				MX6QDL_PAD_SD3_DAT7__UART1_TX_DATA		0x1b0b1
+				MX6QDL_PAD_SD3_DAT6__UART1_RX_DATA		0x1b0b1
+			>;
+		};
+
+		pinctrl_uart2: uart2grp {
+			fsl,pins = <
+				MX6QDL_PAD_SD3_DAT5__UART2_TX_DATA		0x1b0b1
+				MX6QDL_PAD_SD3_DAT4__UART2_RX_DATA		0x1b0b1
+			>;
+		};
+
+		pinctrl_uart3: uart3grp {
+			fsl,pins = <
+				MX6QDL_PAD_EIM_D24__UART3_TX_DATA		0x1b0b1
+				MX6QDL_PAD_EIM_D25__UART3_RX_DATA		0x1b0b1
+			>;
+		};
+
+		pinctrl_uart4: uart4grp {
+			fsl,pins = <
+				MX6QDL_PAD_KEY_COL0__UART4_TX_DATA		0x1b0b1
+				MX6QDL_PAD_KEY_ROW0__UART4_RX_DATA		0x1b0b1
+			>;
+		};
+
+		pinctrl_uart5_1: uart5grp-1 {
+			fsl,pins = <
+				MX6QDL_PAD_KEY_COL1__UART5_TX_DATA	0x1b0b1
+				MX6QDL_PAD_KEY_ROW1__UART5_RX_DATA	0x1b0b1
+				MX6QDL_PAD_KEY_COL4__UART5_RTS_B	0x1b0b1
+				MX6QDL_PAD_KEY_ROW4__UART5_CTS_B	0x1b0b1
+				
+				/* BT_EN wl12xx_bt_en w/ 100K Pull-up  
+					According to the TI wl127x porting guide: 
+					The pullup is needed for keeping the chip enabled when system 
+					is suspending so we would not need to releoad the 
+					firmware after each suspend/resume cycle.
+				*/
+				MX6QDL_PAD_EIM_WAIT__GPIO5_IO00	0x130b0
+			>;
+		};
+
+		pinctrl_uart5dte_1: uart5dtegrp-1 {
+			fsl,pins = <
+				MX6QDL_PAD_KEY_ROW1__UART5_TX_DATA	0x1b0b1
+				MX6QDL_PAD_KEY_COL1__UART5_RX_DATA	0x1b0b1
+				MX6QDL_PAD_KEY_ROW4__UART5_RTS_B	0x1b0b1
+				MX6QDL_PAD_KEY_COL4__UART5_CTS_B	0x1b0b1
+			>;
+		};
+
+		pinctrl_usbotg: usbotggrp {
+			fsl,pins = <
+				MX6QDL_PAD_GPIO_1__USB_OTG_ID		0x17059
+
+				/* USB OTG OC currently not hooked up on claimcheck */
+				/* MX6QDL_PAD_EIM_D21__USB_OTG_OC		0x00000005	 */
+			>;
+		};
+
+		pinctrl_usbh3_1: usbh3grp-1 {
+			fsl,pins = <
+				MX6QDL_PAD_RGMII_RX_CTL__USB_H3_DATA 	0x40013030
+				/* when idle set 100K pull-down */
+				MX6QDL_PAD_RGMII_RXC__USB_H3_STROBE  	0x40013030
+			>;
+		};
+
+		pinctrl_usbh3_2: usbh3grp-2 {
+			fsl,pins = <
+				/* when active enable 47K pull-up */
+				MX6QDL_PAD_RGMII_RXC__USB_H3_STROBE 	0x40017030
+			>;
+		};
+
+		pinctrl_usdhc2: usdhc2grp {
+			fsl,pins = <
+				MX6QDL_PAD_SD2_CMD__SD2_CMD		0x17069
+				MX6QDL_PAD_SD2_CLK__SD2_CLK		0x10069
+				MX6QDL_PAD_SD2_DAT0__SD2_DATA0		0x17069
+				MX6QDL_PAD_SD2_DAT1__SD2_DATA1		0x17069
+				MX6QDL_PAD_SD2_DAT2__SD2_DATA2		0x17069
+				MX6QDL_PAD_SD2_DAT3__SD2_DATA3		0x17069
+				MX6QDL_PAD_GPIO_8__XTALOSC_REF_CLK_32K	0x000b0		/* TiWi slow clock */
+			>;
+		};
+
+		pinctrl_usdhc2_1_100mhz: usdhc2grp-1-100mhz {
+			fsl,pins = <
+				MX6QDL_PAD_SD2_CMD__SD2_CMD		0x170b9
+				MX6QDL_PAD_SD2_CLK__SD2_CLK		0x100b9
+				MX6QDL_PAD_SD2_DAT0__SD2_DATA0		0x170b9
+				MX6QDL_PAD_SD2_DAT1__SD2_DATA1		0x170b9
+				MX6QDL_PAD_SD2_DAT2__SD2_DATA2		0x170b9
+				MX6QDL_PAD_SD2_DAT3__SD2_DATA3		0x170b9
+				MX6QDL_PAD_GPIO_8__XTALOSC_REF_CLK_32K	0x000b0		/* TiWi slow clock */
+			>;
+		};
+
+		pinctrl_usdhc2_1_200mhz: usdhc2grp-1-200mhz {
+			fsl,pins = <
+				MX6QDL_PAD_SD2_CMD__SD2_CMD		0x170f9
+				MX6QDL_PAD_SD2_CLK__SD2_CLK		0x100f9
+				MX6QDL_PAD_SD2_DAT0__SD2_DATA0		0x170f9
+				MX6QDL_PAD_SD2_DAT1__SD2_DATA1		0x170f9
+				MX6QDL_PAD_SD2_DAT2__SD2_DATA2		0x170f9
+				MX6QDL_PAD_SD2_DAT3__SD2_DATA3		0x170f9
+				MX6QDL_PAD_GPIO_8__XTALOSC_REF_CLK_32K	0x000b0		/* TiWi slow clock */
+			>;
+		};
+
+		pinctrl_usdhc3: usdhc3grp {
+			fsl,pins = <
+				MX6QDL_PAD_SD3_CMD__SD3_CMD		0x17059
+				MX6QDL_PAD_SD3_CLK__SD3_CLK		0x10059
+				MX6QDL_PAD_SD3_DAT0__SD3_DATA0		0x17059
+				MX6QDL_PAD_SD3_DAT1__SD3_DATA1		0x17059
+				MX6QDL_PAD_SD3_DAT2__SD3_DATA2		0x17059
+				MX6QDL_PAD_SD3_DAT3__SD3_DATA3		0x17059
+			>;
+		};
+
+		pinctrl_usdhc4: usdhc4grp {
+			fsl,pins = <
+				MX6QDL_PAD_SD4_CMD__SD4_CMD		0x17059
+				MX6QDL_PAD_SD4_CLK__SD4_CLK		0x10059
+				MX6QDL_PAD_SD4_DAT0__SD4_DATA0		0x17059
+				MX6QDL_PAD_SD4_DAT1__SD4_DATA1		0x17059
+				MX6QDL_PAD_SD4_DAT2__SD4_DATA2		0x17059
+				MX6QDL_PAD_SD4_DAT3__SD4_DATA3		0x17059
+				MX6QDL_PAD_SD4_DAT4__SD4_DATA4		0x17059
+				MX6QDL_PAD_SD4_DAT5__SD4_DATA5		0x17059
+				MX6QDL_PAD_SD4_DAT6__SD4_DATA6		0x17059
+				MX6QDL_PAD_SD4_DAT7__SD4_DATA7		0x17059
+			>;
+		};
+
+	};
+};
+
+&dcic1 {
+	dcic_id = <0>;
+	dcic_mux = "dcic-hdmi";
+	status = "okay";
+};
+
+&dcic2 {
+	dcic_id = <1>;
+	dcic_mux = "dcic-lcdif1";
+	status = "okay";
+};
+
+&gpc {
+	/* use ldo-enable, u-boot will check it and configure */
+	fsl,ldo-bypass = <0>;
+	/* use watchdog timer reset output - not connected on claimcheck */
+	/* fsl,wdog-reset = <2>; */
+	/* watchdog select of reset source */
+	fsl,wdog-reset = <1>;
+};
+
+&hdmi_audio {
+	status = "okay";
+};
+
+&hdmi_cec {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hdmi_cec>;
+	status = "okay";
+};
+
+&hdmi_core {
+	ipu_id = <0>;
+	disp_id = <0>;
+	status = "okay";
+};
+
+&hdmi_video {
+	fsl,phy_reg_vlev = <0x0294>;
+	fsl,phy_reg_cksymtx = <0x800d>;
+	HDMI-supply = <&reg_hdmi>;
+	status = "okay";
+};
+
+&ldb {
+	status = "disabled";
+
+	lvds-channel@0 {
+		fsl,data-mapping = "spwg";
+		fsl,data-width = <18>;
+		status = "disabled";
+
+		display-timings {
+			native-mode = <&timing0>;
+			timing0: hsd100pxn1 {
+				clock-frequency = <65000000>;
+				hactive = <1024>;
+				vactive = <768>;
+				hback-porch = <220>;
+				hfront-porch = <40>;
+				vback-porch = <21>;
+				vfront-porch = <7>;
+				hsync-len = <60>;
+				vsync-len = <10>;
+			};
+		};
+	};
+
+	lvds-channel@1 {
+		fsl,data-mapping = "spwg";
+		fsl,data-width = <18>;
+		primary;
+		status = "disabled";
+
+		display-timings {
+			native-mode = <&timing1>;
+			timing1: hsd100pxn1 {
+				clock-frequency = <65000000>;
+				hactive = <1024>;
+				vactive = <768>;
+				hback-porch = <220>;
+				hfront-porch = <40>;
+				vback-porch = <21>;
+				vfront-porch = <7>;
+				hsync-len = <60>;
+				vsync-len = <10>;
+			};
+		};
+	};
+
+};
+
+&mipi_csi {
+	status = "okay";
+	ipu_id = <0>;
+	csi_id = <1>;
+	v_channel = <0>;
+	lanes = <2>;
+};
+
+&pcie {
+	status = "disabled";
+};
+
+&pwm1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm1>;
+	status = "okay";
+};
+
+&ssi2 {
+	fsl,mode = "i2s-slave";
+	status = "okay";
+};
+
+/* Smart Card Reader */
+&uart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart1>;
+	status = "okay";
+};
+
+/* Console */
+&uart2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart2>;
+	status = "okay";
+};
+
+/* GPS */
+&uart3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart3>;
+	status = "okay";
+};
+
+/* NFC */
+&uart4 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart4>;
+	status = "okay";
+};
+
+/* Bluetooth */
+&uart5 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart5_1>;
+	fsl,uart-has-rtscts;
+	status = "okay";
+};
+
+&usbh1 {
+	/* sabresd uses this to turn on the port but actually uses PMIC supply */
+	vbus-supply = <&reg_usb_h1_vbus>; 
+	/* vbus-supply = <&swbst_reg>; */
+	status = "okay";
+};
+
+&usbotg {
+	vbus-supply = <&reg_usb_otg_vbus>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usbotg>;
+	disable-over-current;
+	srp-disable;
+	hnp-disable;
+	adp-disable;
+	status = "okay";
+};
+
+&usbh3 {
+	pinctrl-names = "idle", "active";
+	pinctrl-0 = <&pinctrl_usbh3_1>;
+	pinctrl-1 = <&pinctrl_usbh3_2>;
+	/* created platform driver to control supply regulator */
+	/* vbus-supply = <&reg_usb_h2_vbus>; */
+	disable-over-current;
+	dr_mode = "host";
+	/* tpl-support;	  uses internal whitelist */
+	osc-clkgate-delay = <0x3>;
+	phy-clkgate-delay-us = <400>;
+	status = "okay";
+};
+
+#if 0
+/* this is the PHY for USBH3 */
+&usbphy_nop2 {
+	vcc-supply = <&reg_aux5v>;
+	reset-gpios = <&gpio1 7 GPIO_ACTIVE_LOW>;
+	status = "okay";
+};
+#endif
+
+&usdhc2 {	/* uSDHC2, TiWi wl1271/wl18xx */
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc2>;
+	pinctrl-1 = <&pinctrl_usdhc2_1_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc2_1_200mhz>;
+	bus-width = <4>;
+	non-removable;
+	vmmc-supply = <&reg_wlan_en>;
+	cap-sd-highspeed;
+	cap-mmc-highspeed;
+	//no-1-8-v;
+	//cap-power-off-card;
+	keep-power-in-suspend;
+	wifi-host;
+	//pm-ignore-notify;
+	enable-sdio-wakeup;
+	cap-sdio-irq;
+	status = "okay";
+
+	#address-cells = <1>;
+	#size-cells = <0>;
+	wlcore: wlcore@2 {
+		compatible = "ti,wl1271";
+		reg = <2>;
+		interrupt-parent = <&gpio1>;
+		interrupts = <21 IRQ_TYPE_EDGE_RISING>;
+		platform-quirks = <1>;
+		ref-clock-frequency = <38400000>;
+		tcxo-clock-frequency = <38400000>;
+	};
+};
+
+/* External SD card */
+&usdhc3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc3>;
+	bus-width = <4>;
+	vmmc-supply = <&sw2_reg>;
+	cd-gpios = <&gpio2 0 GPIO_ACTIVE_LOW>;
+	disable-wp;
+	no-1-8-v;
+	keep-power-in-suspend;
+	enable-sdio-wakeup;
+	status = "okay";
+};
+
+/* Internal eMMC card */
+&usdhc4 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc4>;
+	bus-width = <8>;
+	vmmc-supply = <&sw2_reg>;
+	non-removable;
+	no-1-8-v;
+	keep-power-in-suspend;
+	status = "okay";
+};
+
+&wdog1 {
+	status = "okay";
+};
+
+&wdog2 {
+	status = "disabled";
+};
-- 
2.9.5

